### ISSUE TITLE
Fresh install is failing due to GDAL after #287

### ISSUE THREAD

**Issue by richardarsenault** [Ouranosinc/raven#289]
Following the latest merge with the pinned xclim and xarray (#287), RAVEN fails to install. It seems gdal is not installing or failing:

After:
- make clean
- make install

I get the following errors:

 Building wheel for gdal (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/ets/anaconda3/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-fwt5rejb/gdal/setup.py'"'"'; __file__='"'"'/tmp/pip-install-fwt5rejb/gdal/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-9fu1l896 --python-tag cp37
       cwd: /tmp/pip-install-fwt5rejb/gdal/
  Complete output (250 lines):
  running bdist_wheel

[...]

ERROR: Failed building wheel for gdal

And also this after it tries to roll back the uninstall:

ERROR: Command errored out with exit status 1: /home/ets/anaconda3/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-fwt5rejb/gdal/setup.py'"'"'; __file__='"'"'/tmp/pip-install-fwt5rejb/gdal/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-wd4gi007/install-record.txt --single-version-externally-managed --compile Check the logs for full command output.

**Comment by Zeitsperre** [comment]
@richardarsenault I'm nearly convinced that this is related to another issue I raised not long ago (#284). Can you try the following and let us know if it still isn't working?

- `make env_clean`
- `make conda_env`
- `make install`
- `make start`

GDAL from PyPI (and other GIS packages) won't build if you don't have the accompanying C libraries (at appropriate versions) installed, hence why the conda install approach is a much better option.

**Comment by richardarsenault** [comment]
Tried it, and still fails at make install, same problem.

**Comment by tlvu** [comment]
I am on branch `bias_correction` (4fbb8df026e7f61d881b37be9bce91736aca033f), I have a similar problem, gdal problem on `make develop` after a clean `conda env create`.

gdal in environment.yml and requirements.txt do not match, that might be the reason.

environment.yml: gdal~=2.4: https://github.com/Ouranosinc/raven/blob/4fbb8df026e7f61d881b37be9bce91736aca033f/environment.yml#L28

requirements.txt: gdal==3.0.4: https://github.com/Ouranosinc/raven/blob/4fbb8df026e7f61d881b37be9bce91736aca033f/requirements.txt#L45

**Comment by tlvu** [comment]
Oh wait, this branch `bias_correction` is not merged to branch `master` yet.  On `master` gdal in both files are still in sync so not sure what happened.

**Comment by Zeitsperre** [comment]
I'm not sure why it's failing locally for you (seems to be working for me), The RtD build from master was failing, and I have a feeling it could be because I added xclim to the environment.yml dependencies. Will explore more today.

**Comment by tlvu** [comment]
Sorry, just to be clear, I was able to reproduce the gdal problem on branch bias_correction, I did not  try to reproduce on master since I was working on bias_correction.

**Comment by Zeitsperre** [comment]
Yep, all good. I was responding to @richardarsenault. I'm trying to reproduce on master.

**Comment by Zeitsperre** [comment]
Just removed my Raven environment and ran all commands listed (plus `conda activate raven` after `make conda_env`) and all tests are passing. Master is currently still using the older GDAL (~=2.4). Still searching.

**Comment by Zeitsperre** [comment]
@richardarsenault I can't seem to reproduce this error. If GDAL is being properly installed via Anaconda, it shouldn't even try to build a wheel. Is it possible that the `make conda_env` command is silently failing before it tries to install the remaining libraries with pip?

**Comment by richardarsenault** [comment]
I'll try and see what I can do, I'll do a few tests with what you have said above in mind. Thanks for the support!

**Comment by richardarsenault** [comment]
OK I have tried multiple things, the latest is as follows:

1- git checkout master
2- git pull
3- make env_clean
4- make conda_env
5- conda activate raven
6- make install

And it fails with the GDAL problem. Anything else I should be looking at?

**Comment by richardarsenault** [comment]
OK I solved it!

My problem is that (I think) I had a few environments active one over the other, as when I tried "conda deactivate" it cycled through my last envs and eventually hit the "raven" environment again, which was the first one in that chain... not sure how to explain it but anyways, from base conda env I repeated the process and everything worked.

Thanks a lot for the support and sorry for the troubles!
